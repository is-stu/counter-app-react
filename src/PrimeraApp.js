import React from 'react';
import PropTypes from 'prop-types';


export const PrimeraApp = ({ text, subtitle }) => {
  const texto = 'Reforzando el React';
  return (
    <>
      <h1>{texto}</h1>
      <h2>{text}</h2>
      <p>{subtitle}</p>
    </>
  );
};

PrimeraApp.propTypes = {
  text: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
    subtitle: 'Im a subtitle'
}

// El React.Fragment tambien se puede utilizar como <> </> !!!!!!!
// Dado el caso, si se necesita mostrar un obj en {} con la etiqueta <pre></pre>
