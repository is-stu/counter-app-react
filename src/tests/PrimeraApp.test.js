import { shallow } from 'enzyme';
import { PrimeraApp } from '../PrimeraApp';
import React from 'react';
import '@testing-library/jest-dom';

describe('Pruebas en la Primera app', () => {
  //   test('Debe de mostrar el mensaje "Hola soy Goku', () => {
  //     const saludo = 'Hola, soy goku';

  //     const { getByText } = render(<PrimeraApp text={'Hola, soy goku'} />);

  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });

  test('Debe de mostrar PrimeraApp correctamente', () => {
    const saludo = 'Hola, soy goku';
    const wrapper = shallow(<PrimeraApp text={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar el subtitulo enviado por props', () => {
    const saludo = 'Hola, soy goku';
    const subtitle = 'Soy un subtitulo';
    const wrapper = shallow(<PrimeraApp text={saludo} subtitle={subtitle} />);

    const textoParrafo = wrapper.find('p').text();

    expect(textoParrafo).toBe(subtitle);
  });
});
