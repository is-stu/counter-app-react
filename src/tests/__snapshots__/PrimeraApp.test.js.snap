// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la Primera app Debe de mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Reforzando el React
  </h1>
  <h2>
    Hola, soy goku
  </h2>
  <p>
    Im a subtitle
  </p>
</Fragment>
`;
